src/Compilers/VisualBasic/Portable/Binding/TemporaryLookupResults.vb

```diff
@@ -0,0 +1,55 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.PooledObjects
+
+Namespace Microsoft.CodeAnalysis.VisualBasic
+    ''' <summary>
+    ''' TemporaryLookupResults allows the binding process to create an uncontested pool of 
+    ''' LookupResults that can be used by a single thread during binding.  LookupResults are
+    ''' retrieved/returned to this pool as needed, removing the need to get them from the
+    ''' global pool which may be highly contested on high-proc machines.
+    ''' <para/>
+    ''' TemporaryLookupResults instances can be created effectively as high up in the binding stack
+    ''' as is convenient.  Ideally, they are  created at any public entry-point into the binding
+    ''' process and passed along to any code that needs them.
+    ''' </summary>
+    Friend Structure TemporaryLookupResults
```

Comment:
>TemporaryLookupResults [](start = 21, length = 22)

If we pass this structure ByRef, we could allocate _stack lazily on demand and we could keep one LookupResult instance in the field, avoiding allocating/using stack for scenarios when we repeatedly get then free an instance.